sdk/messaging/azservicebus/internal/exported/error.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package exported
+
+import "fmt"
+
+// Code is an error code, usable by consuming code to work with
+// programatically.
+type Code string
+
+const (
+	// CodeConnectionLost means our connection was lost and all retry attempts failed.
+	// This typically reflects an extended outage or connection disruption and may
+	// require manual intervention.
+	CodeConnectionLost Code = "connlost"
+
+	// CodeLockLost means that the lock token you have for a message has expired.
+	// This message will be available again after the lock period expires, or, potentially
+	// go to the dead letter queue if delivery attempts have been exceeded.
+	CodeLockLost Code = "locklost"
+)
+
+// Error represents a Service Bus specific error.
+type Error struct {
```

Comment:
I think we need to decide if this is the path we want to take.  Per the [guidelines](https://azure.github.io/azure-sdk/golang_introduction.html#error-handling) we currently forbid the creation of custom error types; plus, we went to some lengths to update other SDKs to return an `*azcore.ResponseError`.  Having a unified error type across the SDKs simplifies the experience.

CC @JeffreyRichter

Reply:
We had this discussion, actually:
https://github.com/Azure/azure-sdk-for-go/issues/17283#issuecomment-1071013052

Reply:
I didn't see explicitly where it was said that `azcore.ResponseError` won't work, but if you don't think it will then that's fine.

Is there any commonality with event hubs such that they can share an error type, or is it different enough that we'll end up with `azeventhubs.Error`?

Reply:
After discussing we're going to leave this as-is but we want to add in doc comments to indicate where this error type comes back.

Reply:
(doc comments added)